### PR TITLE
Update azure-identity history and version

### PR DESCRIPTION
**Pull request by chlowell**

### CODE REVIEW

File: sdk/identity/azure-identity/azure/identity/aio/_credentials/authorization_code.py [hunk 1]
```diff
@@ -31,13 +32,19 @@ class AuthorizationCodeCredential(object):
...
     def __init__(
-        self,
-        tenant_id: str,
-        client_id: str,
-        authorization_code: str,
-        redirect_uri: str,
-        **kwargs: "Any"
+        self, tenant_id: str, client_id: str, authorization_code: str, redirect_uri: str, **kwargs: "Any"
     ) -> None:
         self._authorization_code = authorization_code  # type: Optional[str]
```
Comment (xiangyan99) [inline sdk/identity/azure-identity/azure/identity/aio/_credentials/authorization_code.py]: Why optional?

File: sdk/identity/azure-identity/tests/test_auth_code_async.py [hunk 2]
```diff
@@ -32,6 +32,32 @@
...
+    credential = AuthorizationCodeCredential(
+        "tenant-id", "client-id", "auth-code", "http://localhost", transport=transport
+    )
+
+    async with credential:
+        assert transport.__aenter__.call_count == 1
+
+    assert transport.__aenter__.call_count == 1
+    assert transport.__aexit__.call_count == 1
+
```
Comment (xiangyan99) [inline sdk/identity/azure-identity/tests/test_auth_code_async.py]: Can you add credential.close() after the context manager to make sure double close does not cause problems?

File: sdk/identity/azure-identity/azure/identity/aio/_credentials/authorization_code.py (hunk 1 above)
Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/aio/_credentials/authorization_code.py]: Note the private field is optional but the parameter is not. The authorization code is a string, so the private field is initially a string. Authorization codes are single use. After successfully redeeming the code for a token, `get_token` sets the private field to `None` to indicate the credential no longer has an authorization code and can only return a cached token. So, `Optional[str]`.

File: sdk/identity/azure-identity/tests/test_auth_code_async.py (hunk 2 above)
Comment (chlowell) [reply sdk/identity/azure-identity/tests/test_auth_code_async.py]: I could, but the credential just wraps the transport's methods, which these tests already verify. It seems like what you really want is to ensure double closing the transport is okay. I want that too, but to me it's a test for the real transport implementations.

**Pull request merged**